enhancements/machine-api/machine-health-check-pause-on-upgrade.md

```diff
@@ -0,0 +1,175 @@
+---
+title: machine-health-check-pause-on-cluster-upgrade
+
+authors:
+  - @slintes
+
+reviewers:
+  - @JoelSpeed
+  - @michaelgugino
+  - @beekhof
+
+approvers:
+  - @JoelSpeed
+  - @michaelgugino
+  - @beekhof
+
+creation-date: 2021-07-14  
+last-updated: 2021-07-14  
+status: implementable  
+
+see-also:
+  - "enhancements/machine-api/machine-health-checking.md"
+---
+
+# Pause MachineHealthChecks During Cluster Upgrades
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Automatically prevent unwanted remediation during cluster upgrades caused by temporarily unhealthy nodes.
+
+## Motivation
+
+During cluster upgrades nodes get temporarily unhealthy, which might trigger remediation when a machineHealthCheck
+resource is targeting those nodes, and they don't get healthy before the configured timeout. This remediation delays
+the cluster upgrade significantly especially on bare metal clusters, and causes needless interruption of customer
+workloads while they are transferred to other nodes.
+
+### Goals
+
+- Automatically pause all machineHealthChecks during cluster upgrades
+- Automatically unpause all machineHealthChecks when cluster upgrade finished
+
+### Non-Goals
+
+tbd
+
+## Proposal
+
+Introduce a new controller, which watches the clusterVersion resource, and pauses all machineHealthChecks when the
```

Comment:
IMO it should also watch for MHCs, to be able to pause newly MHC CRs that might have been created during the upgrade.

Reply:
I'm not sold on this. I can't imagine any scenarios where they would be created automatically

Reply:
not necessarily automatically, could be by an admin.

Reply:
good catch, agree to watch MHCs 👍🏼